transform-plugins/src/main/java/io/cdap/plugin/Normalize.java

```diff
@@ -81,48 +89,85 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
     String[] fieldMappingArray = config.fieldMapping.split(",");
     for (String fieldMapping : fieldMappingArray) {
       String[] mappings = fieldMapping.split(":");
-      Preconditions.checkArgument(mappings.length == 2, "Mapping field '" + mappings[0] + "' is invalid. Both input" +
-        " and output schema fields required.");
-      //Input schema cannot be null, check added for JUnit test case run.
-      if (inputSchema != null) {
-        Preconditions.checkArgument(inputSchema.getField(mappings[0]) != null, "Mapping field '" + mappings[0]
-          + "' not present in input schema.");
+      if (mappings.length != 2) {
+        collector.addFailure("Field name and mapping must be provided.", null)
+          .withConfigElement(NormalizeConfig.FIELD_MAPPING, fieldMapping);
+      } else {
```

Comment:
Avoid adding more layers of nesting when possible
```
if (mappings.length != 2) {
  ...
  continue;
}

...
```

Reply:
Fixed, and in other places as well